src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3753,8 +3741,71 @@ void os::large_page_init() {
     return;
   }
 
-  size_t large_page_size = Linux::setup_large_page_size();
-  UseLargePages          = Linux::setup_large_page_type(large_page_size);
+  // 2) Scan OS info
+  size_t default_large_page_size = scan_default_large_page_size();
+  os::Linux::_default_large_page_size = default_large_page_size;
+  if (default_large_page_size == 0) {
+    // No large pages configured, return.
+    UseTransparentHugePages = false;
+    UseHugeTLBFS = false;
+    UseSHM = false;
```

Comment:
```suggestion
    UseSHM = false;
    if (!FLAG_IS_DEFAULT(UseLargePages)) {
      log_warning(pagesize)("UseLargePages disabled, no large pages configured and available on the system.");
    }
    UseLargePages = false;
```

Reply:
@mgkwill Okay, that lcm assert has been embarrassingly easy to reproduce and fix (even on x64, by just setting default_large_page_size to 0). Suggested change above.

Slight beauty spot here is that we have now the large page warning spread over two places. Maybe you or @kstefanj have a better proposal. Ideally, we would
- scan default large page
- scan available page sizes
- if no default large page or no multiple page sizes or multiple page sizes do not contain default large page, bail out with error.
But I remember we were not sure if the default large page could possibly be missing from the set of large page sizes, so we correct the value further down.

So, it had nothing to do with 64k. Actually I remember trying to fake it once but since the page size goes into the alignment for committing memory, this is not so easy (we fake it on AIX because of complicated boring reasons, but there we have implicit commit on touch).

Cheers, Thomas

Reply:
This part of the init-step is new, before we did not treat a default large page size equal to zero to be a specific problem. It was caught later by the sanity-checks if I'm not mistaken. I wonder if we could do the same now, but I kind of like this being treated separately. 

One thing we could do to make this a little nicer is to add a function:
```
void warn_no_large_pages_configured() {
  if (!FLAG_IS_DEFAULT(UseLargePages)) {
    log_warning(pagesize)("UseLargePages disabled, no large pages configured and available on the system.");
  }
}
```
We could then call that function from both here: 
```
if (default_large_page_size == 0) {
  // No large pages configured, return.
  warn_no_large_pages_configured();
  UseLargePages = false;
  UseTransparentHugePages = false;
  UseHugeTLBFS = false;
  UseSHM = false;
}
```
And at the end of `setup_large_page_type()`:
```
  ...
  warn_no_large_pages_configured();
  return false;
}
```
What do you guys think about that @tstuefe and @mgkwill?

Reply:
> What do you guys think about that @tstuefe and @mgkwill?

Makes sense, I thought the same.